fvm/evm/types/block.go

```diff
@@ -22,6 +23,9 @@ type Block struct {
 	TotalSupply *big.Int
 
 	// ReceiptRoot returns the root hash of the receipts emitted in this block
+	// Note that this value won't be unique to each block, for example for the
+	// case of empty trie of receipts or a single receipt with no logs and failed state
+	// the same receipt root would be reported for block.
```

Comment:
When is this relevant? what kind of transactions cause same receipt root? direct calls?

Reply:
Since we don't have any state root and based on the definition of the receipt hash its going to use the tx result (failed or not failed) as part of the computation. and for example, if two blocks both have a single transaction with failed status and same gas consumption (no logs, etc). they would reference the same receipt root hash.